src/test/java/array/Array1dTest.java

```diff
@@ -0,0 +1,207 @@
+package array;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class Array1dTest {
+
+    @Test
+    void shouldCreateEmptyArrayByConstructor() {
+        // given
+        int size = 10;
+        // when
+        Array1d<Integer> array = new Array1d<>(size);
+        // then
+        assertNotNull(array);
+    }
+
+    @Test
+    void shouldCreateEmptyArrayByFactoryMethod() {
```

Comment:
same - I think only one should be used: constructor or factory method. Constructor is preferable to me